cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraBulkSourceRelation.java

```diff
@@ -107,17 +112,25 @@ private void persist(@NotNull JavaPairRDD<DecoratedKey, Object[]> sortedRDD, Str
     {
         try
         {
-            sortedRDD.foreachPartition(writeRowsInPartition(broadcastContext, columnNames));
+            List<StreamResult> results = sortedRDD
+                                         .mapPartitions(partitionsFlatMapFunc(broadcastContext, columnNames))
+                                         .collect();
+            long rowCount = results.stream().mapToLong(res -> res.rowCount).sum();
+            long totalBytesWritten = results.stream().mapToLong(res -> res.bytesWritten).sum();
+            LOGGER.info("Bulk writer has written {} rows and {} bytes", rowCount, totalBytesWritten);
+            recordSuccessfulJobStats(rowCount, totalBytesWritten);
         }
         catch (Throwable throwable)
         {
+            recordFailureStats(throwable.getMessage());
             LOGGER.error("Bulk Write Failed", throwable);
             throw new RuntimeException("Bulk Write to Cassandra has failed", throwable);
         }
         finally
         {
             try
             {
+                writerContext.publishJobStats();
```

Comment:
I don't think this will do what you think it will do. This object _should_ be treated as a read-only object from the point of view of the executors. Data won't propagate back to the driver once the executors complete

Reply:
So, the change is not propagating data back to the driver, but publishes stats from the executors. I am assuming here that the context is made available to the executors so what you are saying does not need to happen. Let me know if that makes sense.

I have been able to validate this using the in-jvm-dtests.

Reply:
You're right. Fixed to only report/publish stats from the driver.